src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java

```diff
@@ -53,21 +61,30 @@ public static void generateXdocContent() throws Exception {
         for (Path path : templatesFilePaths) {
             final String pathToFile = path.toString();
             final File inputFile = new File(pathToFile);
-            final File outputFile = new File(pathToFile.replace(".template", ""));
-
+            final File tempFile = File.createTempFile(pathToFile.replace(".template", ""), "");
             final XdocsTemplateSinkFactory sinkFactory = (XdocsTemplateSinkFactory)
                     plexus.lookup(SinkFactory.ROLE, XDOCS_TEMPLATE_HINT);
-            final Sink sink = sinkFactory.createSink(outputFile.getParentFile(),
-                    outputFile.getName(), String.valueOf(StandardCharsets.UTF_8));
+            final Sink sink = sinkFactory.createSink(tempFile.getParentFile(),
+                    tempFile.getName(), String.valueOf(StandardCharsets.UTF_8));
             final XdocsTemplateParser parser = (XdocsTemplateParser)
                     plexus.lookup(Parser.ROLE, XDOCS_TEMPLATE_HINT);
             try (Reader reader = ReaderFactory.newReader(inputFile,
                     String.valueOf(StandardCharsets.UTF_8))) {
                 parser.parse(reader, sink);
             }
             finally {
+
```

Comment:
please revert this

Reply:
@AayushSaini101 , please do this

Reply:
@romani Sorry i didn't get you.  Do i need to make a single line ?

Reply:
No , just remove extra empty line

Reply:
@romani Done

Reply:
Not done.
Please remove empty lines at all in finally block

Reply:
Sorry @romani , Done. 
![image](https://github.com/checkstyle/checkstyle/assets/60972989/642d5ac5-7b78-4127-9185-ae134aff1ef6)

Reply:
Resolved